week-9/Homework/mandatory/1-practice/2-code-reading.md

```diff
@@ -61,3 +66,7 @@ console.log(y);
 ```
 
 What will be the output of this code. Explain your answer in 50 words or less.
+//the output will as following:
+9
+{ x: 10 }
+x did not change because it is varibale of primary type, and if we want to change it we need to reassign it ,for example the previous code would change the value of x if we have put x =f1(x);.However for the variable which is not primary type ,we can change it without reassign it as the code above did.
```

Comment:
You'r correct as a primary type variable will pass to the function as a value. However, objects pass by reference. 
This is due to the data structure itself, as in Objects you don't create new stored value but your point to the same reference. For more you can search passing by reference vs by value. And pointer to object.

Reply:
I don't want to be pedantic but it is called primitive type 😛